sdk/src/test/scala/io/horizen/account/state/ContractInteropCallTest.scala

```diff
@@ -16,44 +21,146 @@ class ContractInteropCallTest extends ContractInteropTestBase {
+      val counterInBytesPadded = view.getAccountStorage(contractAddress, COUNTER_KEY)
+      var counter = org.web3j.utils.Numeric.toBigInt(counterInBytesPadded).intValueExact()
+      counter = counter + 1
+      val newValue = org.web3j.utils.Numeric.toBytesPadded(BigInteger.valueOf(counter), 32)
+      view.updateAccountStorage(contractAddress, COUNTER_KEY, newValue)
+      newValue
+    }
+
   }
 
   @Test
   def testNativeContractCallingEvmContract(): Unit = {
     val initialValue = new BigInteger("400000000000000000000000000000000000000000000000000000000000002a", 16)
-    val initialValueHex = "0x" + initialValue.toString(16)
     val initialValueBytes = toUint256Bytes(initialValue)
 
     // deploy the Storage contract (EVM based) and set the initial value
     val storageContractAddress = deploy(ContractInteropTestBase.storageContractCode(initialValue))
+
+    ///////////////////////////////////////////////////////
+    // Test 1: Native contract executes a staticcall on a Smart Contract, calling a readonly function
+    ///////////////////////////////////////////////////////
+
     // call a native contract and pass along the Storage contract address
-    val returnData = transition(getMessage(NativeTestContract.contractAddress, data = storageContractAddress.toBytes))
+    val inputRetrieveRequest = Bytes.concat(BytesUtils.fromHexString(NativeTestContract.STATICCALL_READONLY_TEST_SIG), storageContractAddress.toBytes)
+    var returnData = transition(getMessage(NativeTestContract.contractAddress, data = inputRetrieveRequest))
     // verify that the NativeTestContract was able to call the retrieve() function on the EVM based contract
     assertArrayEquals("unexpected result", initialValueBytes, returnData)
 
     // put a tracer into the context
-    val tracer = new Tracer(new TraceOptions(false, false, false, false, "callTracer", null))
+    var tracer = new Tracer(new TraceOptions(false, false, false, false, "callTracer", null))
     blockContext.setTracer(tracer)
 
     // repeat the call again
-    val returnDataTraced = transition(getMessage(NativeTestContract.contractAddress, data = storageContractAddress.toBytes))
-    // verify that the result is still correct
-    assertArrayEquals("unexpected result", initialValueBytes, returnDataTraced)
+    val returnDataTraced = transition(getMessage(NativeTestContract.contractAddress, data = inputRetrieveRequest))
+    // verify that the result is still correct. Calculate new expected value because before we called inc()
+    var currentExpectedValue = initialValue.add(BigInteger.ONE)
+    var currentExpectedValueBytes = toUint256Bytes(currentExpectedValue)
+    assertArrayEquals("unexpected result", currentExpectedValueBytes, returnDataTraced)
 
-    val traceResult = tracer.getResult.result
-//      println("traceResult" + traceResult.toPrettyString)
+    var traceResult = tracer.getResult.result
+    //println("traceResult" + traceResult.toPrettyString)
```

Comment:
`println` statement unused, same for line 229

Reply:
Done